tools/cli/adminTaskListCommands.go

```diff
@@ -0,0 +1,101 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cli
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+	s "github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common"
+	"github.com/urfave/cli"
+)
+
+// AdminDescribeTaskList shows poller and ack manager related info for a task list.
+func AdminDescribeTaskList(c *cli.Context) {
+	frontendClient := cFactory.ServerFrontendClient(c)
+	domain := getRequiredGlobalOption(c, FlagDomain)
+	taskList := getRequiredOption(c, FlagTaskList)
+	taskListType := s.TaskListTypeDecision
+	if strings.ToLower(c.String(FlagTaskListType)) == "activity" {
+		taskListType = s.TaskListTypeActivity
+	}
+
+	ctx, cancel := newContext()
+	defer cancel()
+	request := &s.DescribeTaskListRequest{
+		Domain:       common.StringPtr(domain),
+		TaskList:     &s.TaskList{Name: common.StringPtr(taskList)},
+		TaskListType: &taskListType,
+		IsAdmin:      common.BoolPtr(true),
+	}
+
+	response, err := frontendClient.DescribeTaskList(ctx, request)
+	if err != nil {
+		ErrorAndExit("Operation DescribeTaskList failed.", err)
+	}
+
+	printAckMgrInfo(response)
+	fmt.Printf("\n")
+
+	pollers := response.Pollers
+	if len(pollers) == 0 {
+		ErrorAndExit(colorMagenta("No poller for tasklist: "+taskList), nil)
+	}
+	printPollerInfo(pollers, taskListType)
+}
+
+func printAckMgrInfo(response *s.DescribeTaskListResponse) {
+	taskIDBlock := response.GetTaskIDBlock()
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetBorder(false)
+	table.SetColumnSeparator("|")
+	table.SetHeader([]string{"Read Level", "Ack Level", "Backlog", "Lease Start TaskID", "Lease End TaskID"})
+	table.SetHeaderLine(false)
+	table.SetHeaderColor(tableHeaderBlue, tableHeaderBlue, tableHeaderBlue, tableHeaderBlue, tableHeaderBlue)
+	table.Append([]string{strconv.FormatInt(response.GetReadLevel(), 10),
+		strconv.FormatInt(response.GetAckLevel(), 10),
+		strconv.FormatInt(response.GetBacklogCountHint(), 10),
+		strconv.FormatInt(taskIDBlock.GetStartID(), 10),
+		strconv.FormatInt(taskIDBlock.GetEndID(), 10)})
+	table.Render()
+}
+
+func printPollerInfo(pollers []*s.PollerInfo, taskListType s.TaskListType) {
```

Comment:
Now that we are using the same types can we use this code to generate a table both in the admin cli and in the public cli?

Reply:
I don't think they have the same type, in admin cli PollerInfo is defined in "github.com/uber/cadence/.gen/go/shared", while in the public cli it's defined in "go.uber.org/cadence/.gen/go/shared". The reason is that, when calling admin DescribeTaskList, we need to use ServerFrontendClient(public cli is using ClientFrontendClient); otherwise, the old(client side) shared.thrift file will be used.

Reply:
Oh yes.